CompuCell3D/core/CompuCell3D/plugins/FocalPointPlasticity/FocalPointPlasticityPlugin.cpp

```diff
@@ -206,436 +184,445 @@ void FocalPointPlasticityPlugin::update(CC3DXMLElement *_xmlData, bool _fullInit
+}
 
+
+std::vector<FocalPointPlasticityTrackerData> FocalPointPlasticityPlugin::getFPPDataVec(CellG * _cell) {
+	std::vector<FocalPointPlasticityTrackerData> dataVec;
+	for each(FocalPointPlasticityLink* link in linkInv.getCellLinkList(_cell))
+		dataVec.push_back(link->getFPPTrackerData(_cell));
+	return dataVec;
 }
 
-std::vector<FocalPointPlasticityTrackerData> FocalPointPlasticityPlugin::getInternalFPPDataVec(CellG * _cell){
-	return std::vector<FocalPointPlasticityTrackerData>
-        (focalPointPlasticityTrackerAccessor.get(
-            _cell->extraAttribPtr)->internalFocalPointPlasticityNeighbors.begin(),focalPointPlasticityTrackerAccessor.get(_cell->extraAttribPtr)->internalFocalPointPlasticityNeighbors.end());
+std::vector<FocalPointPlasticityTrackerData> FocalPointPlasticityPlugin::getInternalFPPDataVec(CellG * _cell) {
+	std::vector<FocalPointPlasticityTrackerData> dataVec;
+	for each(FocalPointPlasticityInternalLink* link in linkInvInternal.getCellLinkList(_cell))
+		dataVec.push_back(link->getFPPTrackerData(_cell));
+	return dataVec;
 }
 
-std::vector<FocalPointPlasticityTrackerData> FocalPointPlasticityPlugin::getAnchorFPPDataVec(CellG * _cell){
-	return std::vector<FocalPointPlasticityTrackerData>(
-        focalPointPlasticityTrackerAccessor.get(
-            _cell->extraAttribPtr)->anchors.begin(),focalPointPlasticityTrackerAccessor.get(_cell->extraAttribPtr)->anchors.end());
+std::vector<FocalPointPlasticityTrackerData> FocalPointPlasticityPlugin::getAnchorFPPDataVec(CellG * _cell) {
+	std::vector<FocalPointPlasticityTrackerData> dataVec;
+	for each (FocalPointPlasticityAnchor* link in linkInvAnchor.getCellLinkList(_cell))
+		dataVec.push_back(link->getFPPTrackerData(_cell));
+	return dataVec;
 }
 
-double FocalPointPlasticityPlugin::tryAddingNewJunction(const Point3D &pt,const CellG *newCell) {
+double FocalPointPlasticityPlugin::tryAddingNewJunction(const Point3D &pt, const CellG *newCell) {
 
-	int currentWorkNodeNumber=pUtils->getCurrentWorkNodeNumber();	
+	int currentWorkNodeNumber = pUtils->getCurrentWorkNodeNumber();
 	short &  newJunctionInitiatedFlag = newJunctionInitiatedFlagVec[currentWorkNodeNumber];
-	CellG * & newNeighbor=newNeighborVec[currentWorkNodeNumber];
-	
-	if (((int)plastParamsArray.size())-1<newCell->type){ 
-        //the newCell type is not listed by the user
-		newJunctionInitiatedFlag=false;
+	CellG * & newNeighbor = newNeighborVec[currentWorkNodeNumber];
+
+	if (((int)plastParamsArray.size()) - 1<newCell->type) { //the newCell type is not listed by the user
+		newJunctionInitiatedFlag = false;
 		return 0.0;
 	}
 
-	if(focalPointPlasticityTrackerAccessor.get(newCell->extraAttribPtr)->focalPointPlasticityNeighbors.size()>=maxNumberOfJunctionsTotalVec[newCell->type]){
-		newJunctionInitiatedFlag=false;
+	//check if new cell can accept new junctions
+	if (linkInv.getNumberOfJunctions(const_cast<CellG*>(newCell)) >= maxNumberOfJunctionsTotalVec[newCell->type]) {
```

Comment:
It looks like we need to keep max number of junctions as a cell attribute. Everythign from FPP tracker data can and sohuld  be moved to separate link object  but maxNUmber of junctions must stay within a cell

Reply:
Or perhaps we can get away with it and it looks like what I mentioned is not necessary